### PR TITLE
Mono repo

### PR DESCRIPTION
**Pull request by nytamin**
This PR contains no functional changes, but changes the file-structure into a mono-repo and moves different parts of the code into different packages.

The new packages are:

* **@mos-connection/connector**: Analogue to the old mos-connection library. Used in Node.js to connect to MOS services
* **@mos-connection/model**: Exposes types and enums. Useful in applications that need to handle the piping of mos-data.
* **@mos-connection/helper**: Exposes functionality for creating MOS-data. Useful for applications that need to create mos-data like a web-client interfase.

This will a breaking change, due to changing the npm-lib name.

TODO:

- [x] Finish the mono-repoification
- [x] Move parts of the code to different packages
- [x] Add tests for exposed stuff in the various packages
- [x] Publish pre-release versions to npm
- [ ] Allow for some testing of the new versions
- [ ] Merge into master
- [ ] Bump version (and clean up changelogs?)
- [ ] Deprecate `mos-connection` on NPM and point towards the new lib names

### CODE REVIEW

**Comment by Julusian** [comment]
@nytamin as this is a breaking change, could we do the proper fix for https://github.com/nrkno/sofie-mos-connection/commit/72aa53a35bc36f479c4db76fbdce26efeed30ca0 and make the DOM field be a string?

File: package.json [hunk 1]
```diff
@@ -1,104 +1,64 @@
...
+		"release:set-version": "lerna version --exact --no-changelog --no-push --yes",
+		"release:bump-release": "lerna version --exact --conventional-commits --conventional-graduate --tag-version-prefix='' --no-push --force-publish=*",
+		"release:bump-prerelease": "lerna version --exact --conventional-commits --conventional-prerelease --tag-version-prefix='' --no-push --force-publish=*",
+		"set-version": "lerna version --exact --no-changelog --no-git-tag-version --no-push --yes",
+		"setup": "lerna bootstrap",
+		"reset": "node scripts/reset.js",
+		"link:all": "lerna exec -- yarn link",
+		"build": "lerna run build --stream",
+		"build:changed": "lerna run build --since head --exclude-dependents --stream",
+		"lint": "lerna exec --parallel --no-bail -- eslint . --ext .ts,.tsx",
```
Comment (Julusian) [inline package.json:22]: This is returning with 'success' when it fails in one of the packages fails.
Resulting in the ci passing when the linter has errors (there are a couple right now)

**Review by Julusian — APPROVED** [review]
Was it intentional to remove typedoc? It is being generated and published as part of the release flow, but will fail as the scripts and dependencies have been removed

Something else to think about is whether jest should be setup to be run on the whole repository at once, rather than setup for each component individually. I did this for my [streamdeck library](https://github.com/julusian/node-elgato-stream-deck), but explicitly avoided it for sofie-core. As this repo is a focussed set of packages, it might also make sense to run jest for it all in one go?
eslint can also be invoked for the whole repository at once if wanted.  
It does mean losing the `yarn test:changed` and `yarn lint:changed` scripts, but means that `yarn lint` and `yarn test` will be a lot nicer to use.

I havent read every file, but tried to skim the structure and read some bits

File: .github/workflows/publish-prerelease.yaml [hunk 2]
```diff
@@ -66,27 +72,34 @@ jobs:
...
+        if: ${{ steps.do-publish.outputs.publish }}
+        run: |
+          yarn build
+        env:
+          CI: true
       - name: Get the Prerelease tag
         id: prerelease-tag
         uses: yuya-takeyama/docker-tag-from-github-ref-action@2b0614b1338c8f19dd9d3ea433ca9bc0cc7057ba
         with:
           remove-version-tag-prefix: false
```
Comment (PeterC89) [inline .github/workflows/publish-prerelease.yaml:85]: Depreciated action

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,13 +1,77 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
+See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
+
+# [2.1.0-alpha.0](https://github.com/nrkno/tv-automation-package-manager/compare/2.0.1...2.1.0-alpha.0) (2022-12-09)
```
[the remark this answers is not in the document]
Comment (nytamin) [reply CHANGELOG.md]: I decided to run the lerna publish prerelease script early to ensure that it works correctly. I'll do some cleanup of the changelogs post-merge.

File: .github/workflows/node.yaml [hunk 4]
```diff
@@ -10,79 +10,73 @@ on:
...
         env:
           CI: true
-      - name: Run tests and generate code coverage
-        if: matrix.node-version == '14.x'
+      - name: Run unit tests
         run: |
-          yarn unit --coverage=true
-        env:
-          CI: true
-      - name: Send coverage
```
[the remark this answers is not in the document]
Comment (nytamin) [reply .github/workflows/node.yaml:69]: re-added now 👍

File: package.json (hunk 1 above)
Comment (nytamin) [reply package.json:23]: Good catch! Removing `--no-bail` fixes the issue

**Pull request merged**